src/GraphQL.Tests/Types/TimeOnlyGraphTypeTests.cs

```diff
@@ -0,0 +1,52 @@
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Globalization;
+using GraphQL.Types;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.Types
+{
+    public class TimeOnlyGraphTypeTests
+    {
+        private readonly TimeOnlyGraphType _type = new();
+
+        [Fact]
+        public void coerces_valid_time()
+        {
+            CultureTestHelper.UseCultures(() =>
+            {
+                var expected = new TimeOnly(10, 12, 01);
+                var input = expected.ToString("O", DateTimeFormatInfo.InvariantInfo);
+
+                var actual = _type.ParseValue(input);
+
+                actual.ShouldBe(expected);
+            });
+        }
+
+        [Fact]
+        public void coerces_invalid_string_to_exception()
+        {
+            CultureTestHelper.UseCultures(() => Should.Throw<FormatException>(() => _type.ParseValue("some unknown time")));
+        }
+
+        [Fact]
+        public void coerces_invalidly_formatted_time_to_exception()
+        {
+            CultureTestHelper.UseCultures(() => Should.Throw<FormatException>(() => _type.ParseValue("11 AM")));
+        }
+
+        [Fact]
+        public void coerces_iso8601_formatted_string_to_time()
+        {
+            CultureTestHelper.UseCultures(() =>
+            {
+                _type.ParseValue("01:02:03.0040000").ShouldBe(new TimeOnly(1, 2, 3, 4));
```

Comment:
Please add more tests with "01:02:03.00400009" and "01:02:03.0040009" values.

Reply:
good catch. only 3 points of milliseconds are supported. i will fix all the FFFFFs https://docs.microsoft.com/en-us/dotnet/api/system.timeonly.-ctor?view=net-6.0#System_TimeOnly__ctor_System_Int32_System_Int32_System_Int32_System_Int32_

Reply:
What do you mean?  This prints 00:00:00.0000001

```cs
Console.WriteLine(new TimeOnly(1).ToString("HH:mm:ss.FFFFFFF"));
```

TimeOnly stores times in ticks: "A time of day expressed in the number of 100-nanosecond units since 00:00:00.0000000."

Reply:
WOW!

Reply:
sorry. i crossed the streams. ignore my previous comment

Reply:
so the difficulty is the readable timeonly constructors do not support anything smaller than milliseconds. u need to use tick overload, resulting in 

```
_type.ParseValue("01:02:03.0040009").ShouldBe(new TimeOnly(37230040009));
```

ugly but it works

Reply:
ok